Loop/Views/InsulinModelSelection.swift

```diff
@@ -0,0 +1,317 @@
+    }
+
+    private func oneUnitBolusEffectPrediction(using model: InsulinModel) -> [GlucoseValue] {
+        let bolus = DoseEntry(type: .bolus, startDate: chartManager.startDate, value: 1, unit: .units)
+        let startingGlucoseSample = HKQuantitySample(type: HKQuantityType.quantityType(forIdentifier: .bloodGlucose)!, quantity: startingGlucoseQuantity, start: chartManager.startDate, end: chartManager.startDate)
+        let effects = [bolus].glucoseEffects(insulinModel: model, insulinSensitivity: viewModel.insulinSensitivitySchedule)
+        return LoopMath.predictGlucose(startingAt: startingGlucoseSample, effects: effects)
+    }
+
+    private var startingGlucoseQuantity: HKQuantity {
+        let startingGlucoseValue = viewModel.insulinSensitivitySchedule.quantity(at: chartManager.startDate).doubleValue(for: glucoseUnit) + glucoseUnit.glucoseExampleTargetValue
+        return HKQuantity(unit: glucoseUnit, doubleValue: startingGlucoseValue)
+    }
+
+    private var endingGlucoseQuantity: HKQuantity {
+        HKQuantity(unit: glucoseUnit, doubleValue: glucoseUnit.glucoseExampleTargetValue)
+    }
+
+    private func isSelected(_ settings: InsulinModelSettings) -> Binding<Bool> {
+        Binding(
+            get: { self.viewModel.insulinModelSettings == settings },
+            set: { isSelected in
+                if isSelected {
+                    withAnimation {
+                        self.viewModel.insulinModelSettings = settings
+                    }
+                }
+            }
+        )
+    }
+
+    private var isWalshModelSelected: Binding<Bool> {
+        Binding(
+            get: { self.viewModel.insulinModelSettings.model is WalshInsulinModel },
+            set: { isSelected in
+                if isSelected {
+                    withAnimation {
+                        self.viewModel.insulinModelSettings = .walsh(WalshInsulinModel(actionDuration: self.viewModel.walshActionDuration))
+                    }
+                }
+            }
+        )
+    }
+
+    var dismissButton: some View {
+        Button(action: dismiss) {
+            Text("Close", comment: "Button text to close a modal")
+        }
+    }
+}
+
+fileprivate struct InsulinModelSettingsRow: View {
+    var title: Text
+    var description: Text
+    @Binding var isSelected: Bool
+
+    var body: some View {
+        HStack(spacing: 0) {
+            VStack(alignment: .leading, spacing: 4) {
+                title
+                    .font(.headline)
+                description
+                    .font(.footnote)
+                    .foregroundColor(.secondary)
+            }
+
+            Spacer(minLength: 12)
+
+            selectionIndicator
+                .frame(width: 26, height: 26)
+        }
+        .animation(nil)
+    }
+
+    @ViewBuilder
+    private var selectionIndicator: some View {
+        if isSelected {
+            Image(systemName: "checkmark.circle.fill")
```

Comment:
[nit] Doesn't the UI design show a plain (non-circled) checkmark (✔️ )?

Reply:
Not based on the spec I'm following here: https://www.figma.com/file/0g6eN2Aee3IfxQeVScXJ3S/Tidepool-Loop-Prototype?node-id=564%3A28694